samples/add-features-with-contingent-values/src/main/java/com/esri/arcgismaps/sample/addfeatureswithcontingentvalues/screens/MainScreen.kt

```diff
@@ -0,0 +1,293 @@
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.arcgismaps.data.CodedValue
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.esri.arcgismaps.sample.addfeatureswithcontingentvalues.R
+import com.esri.arcgismaps.sample.addfeatureswithcontingentvalues.components.MapViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+import com.esri.arcgismaps.sample.addfeatureswithcontingentvalues.components.FeatureEditState
+import kotlin.math.roundToInt
+import kotlinx.coroutines.launch
+
+/**
+ * Main screen layout for the sample app
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MainScreen(sampleName: String) {
+    // create a ViewModel to handle MapView interactions
+    val mapViewModel: MapViewModel = viewModel()
+
+    // flows from view model for displaying state in UI
+    val featureEditState by mapViewModel.featureEditState.collectAsStateWithLifecycle()
+
+    // point on map tapped by user
+    var mapPoint by remember { mutableStateOf<Point?>(null) }
+
+    Scaffold(
+        topBar = { SampleTopAppBar(title = sampleName) },
+        content = {
+            var showBottomSheet by remember { mutableStateOf(false) }
+            val bottomSheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
+
+            Box {
+                MapView(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .padding(it),
+                    arcGISMap = mapViewModel.arcGISMap,
+                    graphicsOverlays = listOf(mapViewModel.graphicsOverlay),
+                    onSingleTapConfirmed = {
+                        it.mapPoint.let { point ->
+                            mapPoint = point
+                            showBottomSheet = true
+                        }
+                    }
+                )
+
+                if (showBottomSheet) {
+                    ModalBottomSheet(
+                        modifier = Modifier.wrapContentHeight(),
+                        sheetState = bottomSheetState,
+                        onDismissRequest = {
+                            showBottomSheet = false
+                            mapViewModel.clearFeature()
+                        }
+                    ) {
+                        val onBottomSheetStateChanged = { state: SheetState ->
```

Comment:
Is there a need for this local variable? This may actually have to be remembered. Alternatively, pass it inline to `BottomSheetContents`:

```
BottomSheetContents(
   ...
   onBottomSheetStateChange = { state: SheetState ->
      if (!state.isVisible) {
         showBottomSheet = false
         mapViewModel.clearFeature()
      }
   },
   ...
)
```

Reply:
Doesn't need to be local rather than inline - I had it that way for clarity. Will change if you think that's better.

No strict need to `remember` it, as it's not getting assigned a new value anywhere else. Could remember it for performance, but I don't think it's complex enough to make any appreciable difference.